src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -14,57 +14,74 @@
 
 package com.google.cloud.tools.appengine.cloudsdk.internal.process;
 
-import static java.lang.ProcessBuilder.Redirect;
-
 import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
 import com.google.cloud.tools.appengine.cloudsdk.process.ProcessOutputLineListener;
 import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
 import com.google.common.base.Charsets;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Scanner;
 
+import static java.lang.ProcessBuilder.Redirect;
+
 
 /**
  * Default process runner that allows synchronous or asynchronous execution. It also allows
  * monitoring output and checking the exit code of the child process.
  */
 public class DefaultProcessRunner implements ProcessRunner {
-  private final boolean async;
-  private final List<ProcessOutputLineListener> stdOutLineListeners;
```

Comment:
I would keep all the "final" modifiers and move the
```
    this.stdOutLineListeners = new ArrayList<>();
    this.stdErrLineListeners = new ArrayList<>();
```
to the first constructor.